hbase-common/src/main/java/org/apache/hadoop/hbase/io/ByteBuffAllocator.java

```diff
@@ -68,10 +68,18 @@
   // default heap allocator, it will just allocate ByteBuffers from heap but wrapped by an ByteBuff.
   public static final ByteBuffAllocator HEAP = ByteBuffAllocator.createOnHeap();
 
+  public static final String ALLOCATOR_ENABLED_KEY = "hbase.server.allocator.enabled";
```

Comment:
We should call it some way which can clear say if this is enabled, it is pooled stuff what we use. Anyways allocator will always be there in RS. It might be HEAP one.

Reply:
Sorry for the confusing, my fault.  let's define the 4 new config keys as: 

```
1. hbase.server.allocator.enabled -> hbase.server.allocator.pool.enabled
2. hbase.server.reservoir.minimal.allocating.size -> hbase.server.allocator.minimal.allocate.size
3. hbase.server.allocator.max.buffer.count
4. hbase.server.allocator.buffer.size
```
Will change all plances.